packages/liveblocks-core/src/connection.ts

```diff
@@ -767,18 +771,31 @@ function createConnectionStateMachine<T extends BaseAuthResult>(
 
   const sendHeartbeat: Target<Context, Event | BuiltinEvent, State> = {
     target: "@ok.awaiting-pong",
-    effect: (ctx: Context) => {
+    effect: (ctx) => {
       ctx.socket?.send("ping");
     },
   };
 
+  const maybeHeartbeat: Target<Context, Event | BuiltinEvent, State> = () => {
+    // If the browser tab isn't visible currently, ask the application if going
+    // zombie is fine
+    const doc = typeof document !== "undefined" ? document : undefined;
+    const canZombie =
+      doc?.visibilityState === "hidden" && delegates.canZombie();
+    return canZombie ? "@idle.zombie" : sendHeartbeat;
```

Comment:
why do we want zombies to not heartbeat?

Reply:
oh, nm, I see that zombie means it's actually disconnected, for some reason in my head I was thinking this was the state when focus was lost but it wasn't disconnected.

Reply:
Correct! Instead of transitioning within the `@ok.*` group to "awaiting pong" state, "going zombie" transitions to the `@idle.zombie` state, which leaves the `@ok.*` state group and as a result the connection will get terminated (the `@ok.*` state group's exit function will handle that).